Content.Server/Singularity/EntitySystems/SingularitySystem.cs

```diff
@@ -67,25 +75,34 @@ public override void Update(float frameTime)
                 }
             }
 
+            var gravityUpdateMultiplier = 0;
             while (_gravityAccumulator > GravityCooldown)
             {
                 _gravityAccumulator -= GravityCooldown;
+                gravityUpdateMultiplier += 1;
+            }
 
-                foreach (var singularity in ComponentManager.EntityQuery<ServerSingularityComponent>())
-                {
-                    Update(singularity, GravityCooldown);
-                }
+            // It's important to note that updates must occur even if gravityUpdateMultiplier == 0.
+            // Otherwise, fast-moving objects such as particles can and do miss the singularity.
+
+            foreach (var singularity in ComponentManager.EntityQuery<ServerSingularityComponent>())
+            {
+                Update(singularity, gravityUpdateMultiplier);
```

Comment:
Particles weren't getting eaten, causing singulo to not work at all.
And stuff like this keeps happening with the singularity. Doing an expensive query is dumb but since stuff keeps on breaking with this...

Reply:
I don't care what the code looked like before, I care what it's going to look like, and doing an expensive query is still dumb.